### PR TITLE
Fix/pps modprobe

### PR DESCRIPTION
**Pull request by rikba**
The PPS time synchronization did not work out of the box on UP Squared Linux 16.04. Fixes:

- Start Kernel module after startup in /etc/rc.local
- Several improvements in installation script in order to rerun the script at a later point

### CODE REVIEW

File: piksi_pps_sync/install/install.sh [hunk 1]
```diff
@@ -1,43 +1,80 @@
...
+  sudo sh -c "echo '# Use USB hotplugging to add new USB devices automatically to the daemon.' >> /etc/default/gpsd"
+  sudo sh -c "echo 'USBAUTO=\"false\"' >> /etc/default/gpsd"
+  sudo sh -c "echo '# Devices gpsd should collect to at boot time.' >> /etc/default/gpsd"
+  sudo sh -c "echo '# They need to be read/writeable, either by user gpsd or the group dialout.' >> /etc/default/gpsd"
+  sudo sh -c "echo 'DEVICES=\"${DEVICE}\"' >> /etc/default/gpsd"
+  sudo sh -c "echo '# Other options you want to pass to gpsd.' >> /etc/default/gpsd"
+  sudo sh -c "echo 'GPSD_OPTIONS=\"-n -r\"' >> /etc/default/gpsd"
+  sudo sh -c "echo '' >> /etc/default/gpsd"
+  sudo sh -c "echo 'GPSD_SOCKET=\"/var/run/gpsd.sock\"' >> /etc/default/gpsd"
+  sudo chmod +x /etc/rc.local
```
Comment (michaelpantic) [inline piksi_pps_sync/install/install.sh]: could use tee for writing the config file (way more readable), made a quick example, see here
https://gist.github.com/michaelpantic/eaa081c78900f97d7a42ce15558f55ec

File: piksi_pps_sync/install/install.sh [hunk 2]
```diff
@@ -1,43 +1,80 @@
...
-sudo sh -c "echo '# GPS + PPS' >> /etc/chrony/chrony.conf"
-sudo sh -c "echo 'refclock PPS /dev/pps0 lock NMEA' >> /etc/chrony/chrony.conf"
-sudo sh -c "echo 'refclock SHM 0 offset 0.02 delay 0.2 refid NMEA' >> /etc/chrony/chrony.conf"
+echo "Do you wish to append a new PPS refclock to /etc/chrony/chrony.conf? [y or Y to accept]"
+read append_chrony_conf
+if [[ $append_chrony_conf == "Y" || $append_chrony_conf == "y" ]]; then
+  sudo sh -c "echo '' >> /etc/chrony/chrony.conf"
+  sudo sh -c "echo '# GPS + PPS' >> /etc/chrony/chrony.conf"
+  sudo sh -c "echo 'refclock PPS /dev/pps0 lock NMEA' >> /etc/chrony/chrony.conf"
+  sudo sh -c "echo 'refclock SHM 0 delay 0.2 refid NMEA' >> /etc/chrony/chrony.conf"
```
Comment (michaelpantic) [inline piksi_pps_sync/install/install.sh]: out of interest, what does the delay param specify?

File: piksi_pps_sync/install/install.sh [hunk 3]
```diff
@@ -1,43 +1,80 @@
...
-sudo sh -c "echo '# Start GPSD' >> /etc/rc.local"
-sudo sh -c "echo 'service gpsd start' >> /etc/rc.local"
-sudo sh -c "echo '' >> /etc/rc.local"
-sudo sh -c "echo 'exit 0' >> /etc/rc.local"
+sudo dpkg-reconfigure gpsd
 
-sudo chmod +x /etc/rc.local
+echo "Do you wish to create and overwrite a new /etc/rc.local? [y or Y to accept]"
+read create_rc_local
+if [[ $create_rc_local == "Y" || $create_rc_local == "y" ]]; then
```
Comment (michaelpantic) [inline piksi_pps_sync/install/install.sh]: rc.local is kind of obsolete and replaced by system.d. But it still works, so I'd leave it like that for now.
I would maybe check if it already exists and append to it or show a warning?

(see https://community.blynk.cc/t/howto-use-systemd-instead-of-rc-local-on-linux-to-auto-start-the-server/20582 for how to replace, but also https://www.claudiokuenzler.com/blog/783/ubuntu-18.04-rc.local-file-not-exist-launch-with-systemd-rc-local about that it still workson 18.04 :-) )

File: piksi_pps_sync/install/install.sh (hunk 2 above)
Comment (michaelpantic) [reply piksi_pps_sync/install/install.sh]: nevermind, looked it up :-)  
https://wiki.alpinelinux.org/wiki/Chrony_and_GPSD

File: piksi_pps_sync/install/install.sh (hunk 3 above)
Comment (rikba) [reply piksi_pps_sync/install/install.sh]: Thanks, I'll change that but I'd need to test it on hardware, so I'll open another PR soon.

**Pull request merged**